### PR TITLE
Cot 809 refactor roles

### PR DESCRIPTION
**Pull request by anthonydummer**
### Jira link (if applicable)
[COT-809](https://tools.hmcts.net/jira/browse/COT-809)

### Change description ###
Re-factored solution to create new joh types rather than re-using existing ones. 

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl [hunk 1]
```diff
@@ -110,6 +110,35 @@ then
   logMsg("Rule : sscs_regional_tribunal_salaried_judge_joh");
 end;
 
+/*
+ * "SSCS Principal Judge-Salaried" business role mapping to JOH.
+ */
+rule "sscs_principal_judge-Salaried_judge_joh"
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl:116]: NB: role name does not match name in log message.

[the remark this answers is not in the document]
Comment (anthonydummer) [reply src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl]: Changes have been made and checked in.

File: src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl [hunk 2]
```diff
@@ -139,6 +168,35 @@ then
   logMsg("Rule : sscs_tribunal_judge_fee_paid_joh");
 end;
 
+/*
+ * "SSCS Judge of the First-tier Tribunal (sitting in retirement)-Fee Paid" business role mapping to JOH.
+ */
+rule "sscs_judge_of_the_first-tier_tribunal_(sitting_in_retirement)_fee_paid_joh"
```
[the remark this answers is not in the document]
Comment (anthonydummer) [reply src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl]: Changes have been made and checked in.

File: src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl [hunk 3]
```diff
@@ -228,6 +286,66 @@ then
...
+      .beginTime($jap.getBeginTime())
+      .endTime($jap.getEndTime())
+      .regionId($jap.getCftRegionIdV1())
+      .baseLocationId($jap.getBaseLocationId())
+      .primaryLocation($jap.getPrimaryLocationId())
+      .contractType($jap.getAppointmentType())
+      .build());
+  $jap.setStatus("JOH_Mapped");
+  update($jap);
+  logMsg("Rule : sscs_chief_medical_member_first-tier_tribunal_salaried_joh");
```
[the remark this answers is not in the document]
Comment (anthonydummer) [reply src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl]: Changes have been made and checked in.

File: src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl [hunk 4]
```diff
@@ -259,6 +377,36 @@ then
   logMsg("Rule : sscs_tribunal_member_disability_fee_paid_joh");
 end;
 
+/*
+ * "SSCS Member of the First-tier Tribunal Lay-Fee Paid" business role mapping to JOH.
+ */
+rule "sscs_member_of_the_first_tier_tribunal_lay-fee_paid_joh"
```
[the remark this answers is not in the document]
Comment (anthonydummer) [reply src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl]: Changes have been made and checked in.

File: src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl [hunk 5]
```diff
@@ -289,6 +437,37 @@ then
   logMsg("Rule : sscs_tribunal_member_financially_qualified_joh");
 end;
 
+/*
+ * "SSCS Member of the First-tier Tribunal" business role mapping to JOH.
+ */
+rule "sscs_member_of_the_first-tier_tribunal-fee_paid_joh"
```
[the remark this answers is not in the document]
Comment (anthonydummer) [reply src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl]: Changes have been made and checked in.

File: src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl [hunk 6]
```diff
@@ -289,6 +437,37 @@ then
...
+      .beginTime($jap.getBeginTime())
+      .endTime($jap.getEndTime())
+      .regionId($jap.getCftRegionIdV1())
+      .baseLocationId($jap.getBaseLocationId())
+      .primaryLocation($jap.getPrimaryLocationId())
+      .contractType($jap.getAppointmentType())
+      .build());
+  $jap.setStatus("JOH_Mapped");
+  update($jap);
+  logMsg("Rule : sscs_member_of_the_first-tier_tribunal-fee_paid_joh");
```
[the remark this answers is not in the document]
Comment (anthonydummer) [reply src/main/resources/validationrules/sscs/sscs-judicial-office-holder-mapping.drl]: Changes have been made and checked in.

**Pull request merged**